fdbclient/ManagementAPI.actor.cpp

```diff
@@ -2959,6 +2959,82 @@ ACTOR Future<Void> acknowledgeBulkLoadTask(Database cx, KeyRange range, UID task
 	return Void();
 }
 
+ACTOR Future<std::vector<RangeLockState>> _getUserRangeLockStates(Transaction* tr, KeyRange range) {
+	if (range.end > normalKeys.end) {
+		throw range_lock_failed();
+	}
+	// FIXME: multiple reads
+	state std::vector<RangeLockState> rangeLockStates;
+	RangeResult result = wait(krmGetRanges(tr, rangeLockPrefix, range));
+	for (int i = 0; i < result.size() - 1; i++) {
+		if (result[i].value.empty()) {
+			continue;
+		}
+		RangeLockState rangeLockState = decodeRangeLockState(result[i].value);
+		rangeLockStates.push_back(rangeLockState);
+	}
+	return rangeLockStates;
+}
+
+ACTOR Future<std::vector<RangeLockState>> getUserRangeLockStates(Database cx, KeyRange range) {
```

Comment:
This API is a bit strange. For a given key range, the return values is a list of locked state, which is not going to be useful for the caller to tell what subrange(s) are locked.